src/org/omegat/core/team2/impl/GITRemoteRepository2.java

```diff
@@ -191,25 +196,49 @@ public String getFileVersion(String file) throws IOException {
 
     protected String getCurrentVersion() throws IOException {
         try (RevWalk walk = new RevWalk(repository)) {
-            Ref localBranch = repository.findRef("HEAD");
-            RevCommit headCommit = walk.lookupCommit(localBranch.getObjectId());
+            RevCommit headCommit = walk.lookupCommit(repository.resolve("HEAD"));
             return headCommit.getName();
         }
     }
 
+    protected String getRemoteDefaultBranchName() throws GitAPIException, ConfigInvalidException, IOException {
+        Ref head = Git.lsRemoteRepository().setRemote(repositoryURL).callAsMap().get("HEAD");
+        if (head != null) {
+           if(head.isSymbolic()) {
+              Ref b = head.getTarget();
+              return b.getName();
+           } else {
+               AnyObjectId id = head.getObjectId();
+               return id.getName();
+           }
+        }
+        return getDefaultBranchName();
+    }
+
+    protected static String getDefaultBranchName()
+            throws ConfigInvalidException, IOException {
+        String branch = SystemReader.getInstance().getUserConfig().getString(
+                ConfigConstants.CONFIG_INIT_SECTION, null,
+                ConfigConstants.CONFIG_KEY_DEFAULT_BRANCH);
+        if (branch == null) {
+            branch = "master";
```

Comment:
When there is no git configuration, fallback to direct value here.
We should check current working branch name of local repository .